tsconfig.json

```diff
@@ -4,8 +4,10 @@
   "compilerOptions": {
     "lib": ["es2022"],
     "outDir": "./lib",
+    "declaration": true,
     "declarationDir": "./lib",
-    "esModuleInterop": true
+    "esModuleInterop": true,
+    "ignoreDeprecations": "5.0"
```

Comment:
what deprecations are we ignoring?

Reply:
Oh I see from the screenshot.

My preference would be to get us in line with non-deprecated typescirpt. So we don't get a weird breakage down the line.

Reply:
this seems to be caused by `"extends": "@tsconfig/node16-strictest/tsconfig.json",` but I'm not totally privy to if that's even necessary or what the node 18+ flavor would be. 

<img width="327" alt="image" src="https://user-images.githubusercontent.com/348464/218857169-2e7a77e9-d0e4-470a-9323-cab37f20ea5e.png">

Reply:
i'll set some time aside to help on this later today...

Reply:
realized it would make more sense to bump from `@tsconfig/node16-strictest` to `@tsconfig/node18` I tried `@tsconfig/node18-strictest` but that was throwing warning/errors between vscode and jest that I couldn't get sorted out in a short amount of time.